### PR TITLE
Add docs for ABFS OAuth support

### PR DESCRIPTION
**Pull request by jirassimok**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive-azure.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Hive Connector with Azure Storage
 =================================
 
-The :doc:`hive` can be configured to query
+The :doc:`hive` connector can be configured to query
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-azure.rst:5]: This link adds Hive Connector automatically... 

If you want to control the text you can use

```
:doc:`Hive connector <hive>`
```

File: presto-docs/src/main/sphinx/connector/hive-azure.rst [hunk 2]
```diff
@@ -31,12 +31,14 @@ configured independently with the storage credentials, you can use the following
 properties in the catalog configuration.
 
 We suggest to use this kind of configuration when you only have the Primary
-Storage account linked to the cluster. When there are secondary storage accounts
-involved, it is recommended to configure Presto using
-``hive.config.resources=<path_to_hadoop_core-site.xml>``, provided
-``core-site.xml`` has the all storage account credentials.
+Storage account linked to the cluster. When there are secondary storage
+accounts involved, we recommend configuring Presto using a ``core-site.xml``
+containing the appropriate credentials for each account, as described above.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-azure.rst:36]: as described in the preceding section.

File: presto-docs/src/main/sphinx/connector/hive-azure.rst [hunk 3]
```diff
@@ -47,21 +49,47 @@ involved, it is recommended to configure Presto using
   * - ``hive.azure.wasb-access-key``
     - The decrypted access key for the Azure Blob Storage
 
-If you choose to use ADLS Gen2, you need to add the following properties:
+ADLS Gen2 / ABFS Storage
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To connect to ABFS storage, you may either use the storage account's access
+key or a service principal. Do not use both sets of properties at the
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-azure.rst:56]: key, or

File: presto-docs/src/main/sphinx/connector/hive-azure.rst (hunk 1 above)
Comment (jirassimok) [reply presto-docs/src/main/sphinx/connector/hive-azure.rst]: Oops; fixed.

File: presto-docs/src/main/sphinx/connector/hive-azure.rst [hunk 4]
```diff
@@ -47,21 +50,47 @@ involved, it is recommended to configure Presto using
...
+    - Description
+  * - ``hive.azure.abfs.oauth.endpoint``
+    - The service principal / application's OAuth 2.0 token endpoint (v1).
+  * - ``hive.azure.abfs.oauth.client-id``
+    - The service principal's client/application ID.
+  * - ``hive.azure.abfs.oauth.secret``
+    - A client secret for the service principal.
+
+When using a service principal, it must have the Storage Blob Data Owner,
+Contributor, or Reader role on the storage account you are using, depending on
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-azure.rst:85]: depending on the planned operations.

**Review by mosabua — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**